File: src/main/java/sprykerFileCreator/model/resolver/ClassResolverInterface.java
package sprykerFileCreator.model.resolver;

import pav.sprykerFileCreator.model.renderer.dto.PhpClassInterface;

public interface ClassResolverInterface {

    PhpClassInterface resolveBundleClass(String classType, String bundleName) throws Exception;

    PhpClassInterface resolveClass(String fullQualifiedName, Boolean findInterface, Boolean findClassBelow);

    PhpClassInterface resolveClassBelow(String fullQualifiedName);

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/services/resolver/ClassResolverInterface.kt
package com.github.tobi812.sprykerplugin.services.resolver

import com.github.tobi812.sprykerplugin.services.renderer.dto.PhpClassInterface
import java.lang.Exception

interface ClassResolverInterface {
    @Throws(Exception::class)
    fun resolveBundleClass(classType: String, bundleName: String): PhpClassInterface?
    fun resolveClass(fullQualifiedName: String, findClassBelow: Boolean, findInterface: Boolean): PhpClassInterface?
    fun resolveClassBelow(fullQualifiedName: String): PhpClassInterface?
}
